storage/csi-snapshot.md

```diff
@@ -0,0 +1,120 @@
+---
+title: CSI Snapshots
+authors:
+  - "@chuffman"
+reviewers:
+  - "@gnufied”
+  - “@jsafrane”
+approvers:
+  - "@..."
+creation-date: 2019-09-05
+last-updated: 2019-09-25
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# csi-snapshots
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs] 
+
+## Summary
+We want to include the upstream CSI Snapshot feature in OpenShift.
+
+## Motivation
+
+* This feature has been requested by both the OCS and CNV teams.
+* Snapshotting is an incredibly useful function when it comes to system changes in development areas, as it allows rapid rollback to preview dev versions
+* Container development benefits from filesystems that provide snapshot functionality
+
+### Goals
+
+* Backport and include the external-snapshotter sidecar.
+
+* Package and ship a downstream image of the this sidecar.
+
+### Non-Goals
+
+* Offer application-consistent snapshots by providing pre/post snapshot hooks to freeze/unfreeze applications and/or unmount/mount file system.
+
+* Provide higher-level management, such as backing up and restoring a pod and statefulSet, and creating a consistent group of snapshots.
```

Comment:
I'm not sure if I follow thie non-goals. Are these from the upstream KEP?

Would it make sense to remove them?

Reply:
They were from the upstream KEP. I removed them here.